index.js

```diff
@@ -411,7 +411,8 @@
         Z.equals (this.name, other.name) &&
         Z.equals (this.url, other.url) &&
         Z.equals (this.supertypes, other.supertypes) &&
-        Z.equals (this.keys, other.keys) &&
+        this.keys.length === other.keys.length &&
+        this.keys.every (function(k) { return other.keys.includes (k); }) &&
```

Comment:
This assumes that `this.keys` and `other.keys` are set-like arrays. I believe this to be a safe assumption.